cli/src/main/java/io/quarkiverse/tekton/cli/pipeline/PipelineExec.java

```diff
@@ -68,63 +62,29 @@ public void process(List<HasMetadata> resources) {
             output.out().println("Executing Pipeline: " + pipelineName);
             HasMetadata t = getInstalledPipeline(pipelineName).get();
 
-            List<WorkspaceBinding> workspaceBindings = new ArrayList<>();
-            Map<String, String> parameters = new HashMap<>();
-
-            if (t instanceof io.fabric8.tekton.v1beta1.Pipeline v1beta1Pipeline) {
-                v1beta1Pipeline.getSpec().getWorkspaces().forEach(w -> {
-                    String workspaceName = w.getName();
-
-                    WorkspaceBindings.forName(projectName, workspaceName)
-                            .or(() -> !Boolean.TRUE.equals(w.getOptional())
-                                    ? WorkspaceBindings.forEmpty(projectName, workspaceName)
-                                    : Optional.empty())
-                            .ifPresent(workspaceBindings::add);
-                });
-                v1beta1Pipeline.getSpec().getParams().forEach(p -> parameters.put(p.getName(), p.getType()));
-            } else if (t instanceof io.fabric8.tekton.v1.Pipeline v1Pipeline) {
-                v1Pipeline.getSpec().getWorkspaces().forEach(w -> {
-                    String workspaceName = w.getName();
-
-                    WorkspaceBindings.forName(projectName, workspaceName)
-                            .or(() -> !Boolean.TRUE.equals(w.getOptional())
-                                    ? WorkspaceBindings.forEmpty(projectName, workspaceName)
-                                    : Optional.empty())
-                            .ifPresent(workspaceBindings::add);
-                });
-                v1Pipeline.getSpec().getParams().forEach(p -> parameters.put(p.getName(), p.getType()));
-            }
-
-            List<Param> params = parameters.size() == 1
-                    ? Params.createSingle(parameters.keySet().iterator().next(), parameters.values().iterator().next(),
-                            pipelineArgs)
-                    : Params.create(pipelineArgs);
+            // Get the PipelineRun generated by the TektonProcessor
+            PipelineRun pipelineRun = resources.stream()
+                    .filter(PipelineRun.class::isInstance)
+                    .map(PipelineRun.class::cast)
+                    .findFirst()
+                    .orElseThrow(() -> new IllegalStateException("No PipelineRun found"));
 
-            for (WorkspaceBinding binding : workspaceBindings) {
-                WorkspaceBindings.createIfNeeded(binding);
-            }
+            // Convert the arguments passed by the Cli command as List<String> into a Map<String,String> where the key is equal is the left part of key=val
+            List<Param> params = Params.create(pipelineArgs.stream()
```

Comment:
I think that we should better move the splitting logic inside the `Params.create()` method, so that we have the same behavior everywhere. This would also make testing easier.

Reply:
Fixed: 02c3637